libs/langchain/langchain/agents/openai_assistant/base.py

```diff
@@ -621,9 +621,10 @@ async def _aparse_intermediate_steps(
     ) -> dict:
         last_action, last_output = intermediate_steps[-1]
         run = await self._wait_for_run(last_action.run_id, last_action.thread_id)
-        required_tool_call_ids = {
-            tc.id for tc in run.required_action.submit_tool_outputs.tool_calls
-        }
+        if run.required_action:
+            required_tool_call_ids = {
+                tc.id for tc in run.required_action.submit_tool_outputs.tool_calls
+            }
```

Comment:
In the case when `required_action` is `None`, doesn't that leave `required_tool_call_ids` unbound, which will result in an error on line 631:

```python
if action.tool_call_id in required_tool_call_ids
```
in the list comprehension?

Reply:
You're right. Just added code to fix that error - do you mind adding a test case and I can review and then merge?